src/SPIFFSLogger.h

```diff
@@ -166,7 +166,7 @@ size_t SPIFFSLogger<T>::readRowsBetween(SPIFFSLogData<T> *output, time_t fromTim
                 hits++;
                 if (hits > startIdx)
                 {
-                    SPIFFSLogData<T> *curOutput = output + (copied * sizeof(SPIFFSLogData<T>));
+                    SPIFFSLogData<T> *curOutput = output + copied;
```

Comment:
What the original code is doing here is calculating the memory address that the new result should be copied to by adding an offset to the base `output` address.

While pointer arithmetic is a _bad thing_<sup>TM</sup>, this does work as intended. As an example, if `output` pointed to `0x0` and `sizeof(SPIFFSLogData<T>)` was 8, we'd have:

- 1st result (i.e. `copied == 0`): `curOutput == 0x00`
- 2nd result:  `curOutput == 0x08`
- 3rd result: `curOutput == 0x10`

And so on... Your proposed change actually introduces a bug by assuming that `sizeof(SPIFFSLogData<T>) == 1` which will never be true, and would therefore make us overwrite data.